File: examples/json/base/commons.py
def assert_dict(a, b):
    # to improve
    return a == b


def assert_list(a, b):
    # won't work with nested lists
    asserts = []
    assert len(a) == len(b)
    for index, item in enumerate(a):
        b_item = b[index]
        asserts.append(item == b_item and type(item) == type(b_item))
    return all(asserts)


def assert_float(a, b):
    return isinstance(a, float) and isinstance(b, float) and a == b

File: examples/json/base/commons.js
const assert = require('assert');

const assert_dict = (a, b) => {
    try {       
        assert.deepStrictEqual(a, b);
        return true;
    } catch (error) {
        return false
    }
}

const assert_list = (a, b) => {
    try {       
        assert.deepStrictEqual(a, b);
        return true;
    } catch (error) {
        return false
    }
}

const assert_float = (a, b) => {
    return (
        typeof a === "number" && typeof b === "number" && !Number.isInteger(a) && !Number.isInteger(b) && a === b
    )
}

exports.assert_float = assert_float
exports.assert_list = assert_list
exports.assert_dict = assert_dict
